File: FMAplication/ClientApp/src/app/Shared/Entity/Daily-Activity/audit-report.ts
import { DailyCMActivity } from '.';

export class AuditReport {
    public  id : number; 
    public  dailyAuditId : number;      
    public  dailyCMActivityId : number;
    public  productId : number;
    public  posmProductId : number;
    public  productName : string;
    public  amount : number;
    public  actionType : number;
    
    public  uploadedImageUrl1 : string;
   
    public  uploadedImageUrl2 : string;
    public  status : number; 
    public  productImageUrl : string;

    public dailyCMActivity: DailyCMActivity;

    //display

    public displayStatus: string;
    public displayActionType: string;

    public displayDate : string;
    public fmUserName : string;
    public cmUserName : string;
    public outletName : string;
    public salesPointName : string;
}

File: FMAplication/Domain/Reports/AuditReport.cs
using FMAplication.Core;
using FMAplication.Domain.DailyActivities;
using FMAplication.Domain.Products;
using FMAplication.Enumerations;
using System.ComponentModel.DataAnnotations;

namespace FMAplication.Domain.Reports
{

    public class AuditReport : AuditableEntity<int>
    {
        public int DailyAuditId { get; set; }        
        public int DailyCMActivityId { get; set; }
        public int? ProductId { get; set; }
        public int? POSMProductId { get; set; }
        public int Amount { get; set; }
        public ActionType ActionType { get; set; }
        //[StringLength(512)]
        public string UploadedImageUrl1 { get; set; }
        //[StringLength(512)]
        public string UploadedImageUrl2 { get; set; }
        public Product Product { get; set; }
        public POSMProduct POSMProduct { get; set; }
        public DailyCMActivity DailyCMActivity { get; set; }
    }
}
